### PR TITLE
Compile base-asm to Java6 bytecode

### PR DESCRIPTION
**Pull request by asodja**
To fix running TestNG tests with Java6 and Java7.

Fixes https://github.com/gradle/gradle-private/issues/4281

### CODE REVIEW

**Comment by asodja** [comment]
@bot-gradle test this

**Comment by asodja** [comment]
@bot-gradle test this

File: platforms/core-runtime/base-asm/src/main/java/org/gradle/model/internal/asm/BytecodeFragment.java [hunk 1]
```diff
@@ -20,7 +20,11 @@
 
 public interface BytecodeFragment {
 
-    BytecodeFragment NO_OP = visitor -> {};
+    @SuppressWarnings("Convert2Lambda")
```
Comment (lptr) [inline platforms/core-runtime/base-asm/src/main/java/org/gradle/model/internal/asm/BytecodeFragment.java:23]: ❓ Why is this needed if the code is Java 6 only?

File: platforms/core-runtime/base-asm/src/main/java/org/gradle/model/internal/asm/ClassVisitorScope.java [hunk 2]
```diff
@@ -106,10 +106,16 @@ protected void addGetter(String methodName, Type returnType, String methodDescri
     /**
      * Adds a getter that returns the value that the given code leaves on the top of the stack.
      */
-    protected void addGetter(String methodName, Type returnType, String methodDescriptor, @Nullable String signature, BytecodeFragment body) {
-        publicMethod(methodName, methodDescriptor, signature, methodVisitor -> new MethodVisitorScope(methodVisitor) {{
-            emit(body);
-            _IRETURN_OF(returnType);
-        }});
+    @SuppressWarnings("Convert2Lambda")
```
Comment (lptr) [inline platforms/core-runtime/base-asm/src/main/java/org/gradle/model/internal/asm/ClassVisitorScope.java:109]: ❓ Or this..?

File: platforms/core-runtime/base-asm/src/main/java/org/gradle/model/internal/asm/BytecodeFragment.java (hunk 1 above)
Comment (asodja) [reply platforms/core-runtime/base-asm/src/main/java/org/gradle/model/internal/asm/BytecodeFragment.java]: Yes, not needed, fixed

File: platforms/core-runtime/base-asm/src/main/java/org/gradle/model/internal/asm/ClassVisitorScope.java (hunk 2 above)
Comment (asodja) [reply platforms/core-runtime/base-asm/src/main/java/org/gradle/model/internal/asm/ClassVisitorScope.java]: Indeed, not needed, fixed

**Comment by asodja** [comment]
@bot-gradle test this
